File: edutech_frontend/src/types/lesson.ts
export interface Category {
  id: number;
  name: string;
  description?: string;
  user_id?: number;
  created_at?: string;
  updated_at?: string;
}

export interface CategoryCreate {
  name: string;
  description?: string;
}

export interface CategoryUpdate {
  name?: string;
  description?: string;
}

export interface Quiz {
  id: number;
  title: string;
  description?: string;
  lesson_id: number;
  created_at?: string;
  updated_at?: string;
}

export interface QuizQuestion {
  id: number;
  quiz_id: number;
  question: string;
  options: string[];
  correct_answer: string;
  explanation?: string;
  lesson_id?: number;
  created_at?: string;
  updated_at?: string;
}

export interface Lesson {
  id: number;
  title: string;
  content: string;
  summary?: string;
  description?: string;
  category_id?: number;
  category?: Category;
  user_id: number;
  created_at: string;
  updated_at: string;
}

export interface LessonCreate {
  title: string;
  content: string;
  summary?: string;
  description?: string;
  category_id?: number;
}

export interface LessonUpdate {
  title?: string;
  content?: string;
  summary?: string;
  description?: string;
  category_id?: number;
}

export interface Note {
  id: number;
  content: string;
  selected_text: string;
  lesson_id: number;
  user_id: number;
  created_at: string;
  updated_at: string;
}

File: edutech_backend/app/models/lesson.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime, func, Text
from sqlalchemy.orm import relationship
from app.database import Base

class Lesson(Base):
    __tablename__ = "lessons"
    
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    category_id = Column(Integer, ForeignKey("categories.id"), nullable=True)
    title = Column(String(200), nullable=False)
    content = Column(Text, nullable=True)
    summary = Column(Text, nullable=True)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    
    # Relationships
    user = relationship("User", back_populates="lessons")
    category = relationship("Category", back_populates="lessons")
    
    # Fix: Specify foreign_keys to resolve ambiguity
    documents = relationship(
        "Document", 
        back_populates="lesson", 
        foreign_keys="Document.lesson_id",
        cascade="all, delete-orphan"
    )
    
    # Separate relationship for converted documents
    converted_documents = relationship(
        "Document",
        foreign_keys="Document.converted_lesson_id",
        cascade="all, delete-orphan"
    )
    
    notes = relationship("Note", back_populates="lesson", cascade="all, delete-orphan")
    highlights = relationship("Highlight", back_populates="lesson", cascade="all, delete-orphan")
    quizzes = relationship("Quiz", back_populates="lesson", cascade="all, delete-orphan")